scipy/stats/tests/Vogelsang_trend_test.py

```diff
@@ -0,0 +1,151 @@
+import sys
+import numpy as np
+import pandas as pd
+from numpy.linalg import inv
+
+def vogelsang_trend(y):
+    """
+    Computes the individual tests using the t-PS test of Vogelsang 
+    (1998, Econometrica)
+    
+    Test the null hypothesis that a time series with serial correlation has a 
+    slope with statistically significant trend.
+    
+    The test is evaluated at 90%, 95%, 97.5%, and 99.0% confidence levels.
+    
+    Parameters
+    ----------
+    y : array-like 
+        The time series for which the trend is to be calculated and tested for 
+        statistical significance.
+        
+    Returns
+    -------
+    slope: float
+        The slope of the trend function
+    pvalue: binary 
+        Binary result (1 or 0) for pass or fail for 4 levels of statistical 
+        significance at: 90%, 95%, 97.5%, and 99.0% confidence levels.
+        
+    See Also
+    --------
+    https://www.jstor.org/stable/2998543
+    
+    Notes
+    -----
+    This test is applicable when the time series has a serial auto correlation. 
+    which in mann-kendall and modified mann-kendall test will have a tendency 
+    for over rejection.
+    
+    This procedure computes the nonparametric estimator of the long run variance 
+    using Andrews (1991) AR(1) plug in optimal truncation lag.  
+    v is the vector of residuals and M is the trucation lag.  
+    M<0 uses the automatic bandwidth.  
+    Kernel is an integer representing the kernel that is used 1 = bartlett, 
+    2 = parzen, 3 = Tukey-Hanning, 4 = quadratic spectral
+    if prewhite == 1 then prewhitening using an AR(1) model is used
+    if prewhite == 0 no prewhitening is used.
+
+    References
+    ----------
+    Vogelsang, Timothy J. 1998. “Trend Function Hypothesis Testing in the 
+    Presence of Serial Correlation.” Econometrica 66 (1): 123–48. 
+    https://doi.org/10.2307/2998543.
+    
+    """
+    
+    print("\n###############################################\n")
+    print("Trend Function Hypothesis Test (Vogelsang, 1998)")
+    print("https://www.jstor.org/stable/2998543")
+    print("\n###############################################\n")
+
+    # Critical values for right tailed 5%, 2.5%, 1% test and one-sided
+    # Confidence intervals
+
+    cvtps = [1.331, 1.720, 2.152, 2.647]
+    btps = [0.494, 0.716, 0.995, 1.501]
+    
+    z = np.cumsum(y)    
+    T = len(y)
+    t1 = range(1,T+1)
+    X1 = pd.DataFrame({"T":[1]*T, "t1":t1})
+
+    # Compute the partial sums of X1
+    nr = X1.shape[0] #nc = X1.shape[1]
+    X2 = np.cumsum(X1)
+    
+    # Compute inverse matrices
+    X1inv = inv(np.matmul(X1.T, X1))
+    X2inv = inv(np.matmul(X2.T, X2))
+    
+    # Compute orthonormal trends for the J statistics
+    Xj = np.array(X1["T"] / np.sqrt(nr))
+    Xj = pd.DataFrame(Xj)
+    i = 1;
+    for i in range(1,10):
+        tt = np.array(t1)**i
+        ehat = tt - np.matmul(Xj, np.matmul( np.matmul(Xj.T, Xj), 
```

Comment:
Good catch! Yes, the line should be 
       
```suggestion
         ehat = tt - np.matmul(Xj, np.matmul(inv(np.matmul(Xj.T, Xj)), 
                                            np.matmul(Xj.T,tt)))
```